### PR TITLE
go: Use the new runtime ID allocation scheme

### PR DESCRIPTION
**Pull request by Yawning**
* [x] Use the new runtime ID allocation scheme.
 * [x] Document the new runtime ID allocation scheme.
 * [x] Write one of those stupid changelog fragments.

### CODE REVIEW

File: go/oasis-node/cmd/registry/runtime/runtime.go [hunk 1]
```diff
@@ -260,8 +260,20 @@ func runtimeFromFlags() (*registry.Runtime, signature.Signer, error) {
 			}
 			kmID = &tmpKmID
 		}
+		if id.IsKeyManager() {
+			logger.Error("runtime ID does has the key manager flag set",
```
Comment (kostko) [inline go/oasis-node/cmd/registry/runtime/runtime.go]:
```suggestion
			logger.Error("runtime ID has the key manager flag set",
```

File: go/registry/api/api.go [hunk 2]
```diff
@@ -1005,6 +1031,10 @@ type ConsensusParameters struct {
 	// allowed outside of the genesis block.
 	DebugAllowRuntimeRegistration bool `json:"debug_allow_runtime_registration"`
 
+	// DebugAllowTestRuntimes is true iff test runtimes should be allowed to
+	// be registered.
+	DebugAllowTestRuntimes bool `json:"debug_allow_test_runtimes"`
```
Comment (kostko) [inline go/registry/api/api.go]: Require "don't blame oasis"?

Comment (Yawning) [reply go/registry/api/api.go]: I can see external developers using this flag, unlike a lot of the other ones that are feature gated that way.  If we are serious about having third parties writing runtimes, it is probably best not to feature gate it that way?

Comment (kostko) [reply go/registry/api/api.go]: Yeah, makes sense.

**Pull request merged**